### ISSUE TITLE
Unable to Copy Just the Link from a Message

### ISSUE THREAD

**Issue by wesdawg** [WhisperSystems/Signal-Android#6343]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
I can copy an entire message and open any URL links in a message, but am unable to copy just the link from a message.

### Steps to reproduce
- Compose and send a message that contains an https URL link.

**Actual result:** I can click the link to visit it in a browser, or touch and hold the message to open the menu option to copy the entire message, but there doesn't appear to be a way in either the message details or the conversation view to just copy the link without opening it or copying the entire message.
**Expected result:** Expecting a way to copy just a link, ideally a touch and hold context menu option to copy the link from a message.

### Device info
**Device:** Samsung Galaxy
**Android version:** 6.0.1
**Signal version:** 3.30.4

**Comment by FeuRenard** [comment]
WhatsApp copies the link directly without showing a context menu. There is just a toast to indicate what happened.

**Comment by R4md4c** [comment]
Hi, I'm interested in fixing this issue.

I did a PoC on Signal by linkifying the SpannableString returned from MessageRecord.getDisplayBody(), and then replacing the URLSpan with a custom Span that I wrote and it is working quite well unless there is something that I'm unaware of. 

Now, I have a question regarding the UX part. When the user clicks on a link I send that Uri to the Clipboard manager and then show a Toast with text "Copied %s to clipboard".
Is that acceptable behavior from your point of view? or should I display a context menu with something like "Copy" and "Open Link"?
